### ISSUE TITLE
UX issue with opening beatmap pages

### ISSUE THREAD

**Issue by Hiviexd** [ppy/osu-web#10355]
this issue has been bothering me for years so i thought it'd give reporting it a shot

i have my default mode set as taiko, however when i click on a standard-only beatmap from the beatmap listing/profile page, it opens the page but on the taiko convert portion of the page, which gets really annoying having to manually switch back to standard every time, because when someone clicks on a standard-only map, they mostly intend to access the standard portion of the page.

i can see the current logic working for hybrids where there's proper taiko diffs in the set, but when a set has only diffs from 1 mode, that mode should be treated as primary on the beatmap page

**Comment by peppy** [comment]
Still hoping for more opinions on this, I'm neutral.

**Comment by tailsdk** [comment]
Yeah it would be nice if it stayed on standard. There are more people playing multiple modes then there are people playing converts for the most part

**Comment by kj415j45** [comment]
It's natural to open the convert diff from profile page (or specifically: from score).

![image](https://github.com/ppy/osu-web/assets/18349191/c1e78eea-c084-4acf-ba97-53308b5ad2df)

But for beatmapsets' card like in beatmap listing and profile favourite. How about open convert when clicking on the set itself (current behavior), and open origin mode when clicking on the diff.

![UX preview of my opinion.](https://github.com/ppy/osu-web/assets/18349191/10f9c812-e8c8-4a8a-8e64-ca7872374de5)

**Comment by tailsdk** [comment]
Maybe a toggle would be better as a setting because like almost nobody in mania is looking for the mania convert when clicking on a standard set

**Comment by respektive** [comment]
I think adding a checkbox in the account settings, in the options section next to the other beatmap options, would work well. something like this maybe
![image](https://github.com/ppy/osu-web/assets/22381687/c003d476-c8de-4b6b-9f2d-14309bf06eef)

**Comment by cl8n** [comment]
from @cesarila

> As a former osu!standard main that is now an osu!mania enjoyer, I wanted to set my default game mode to osu!mania to reflect my more recent activities. However, I still talk with people about standard regularly, and was surprised to see that looking up standard mapsets to recommend to people on the website defaulted me to the mania converts.
> 
> Basically, I want to set mania as my default game mode, and have beatmap listings show up in their intended game mode. In cases of listings with multiple game modes represented, something like the following could be used to decide what game mode to show:
> 
> 1. Default game mode, if any non-converted difficulties exist in that game mode
> 2. Whichever game mode has the most non-converted difficulties otherwise
